### PR TITLE
Thank donors on first sign in (2nd attempt)

### PR DESCRIPTION
**Pull request by bencodeorg**
Reintroduces https://github.com/code-dot-org/code-dot-org/pull/35914, which was reverted in code-dot-org/code-dot-org#36169 -- additional changes included in this PR:

- Change add viewport height requirement for showing dialog -- only show on screens where entire dialog will fit (not currently responsive).
- Adds a UI test option `@only_phone`, which runs a scenario only on our phone simulator.
- Fix UI tests that fail when run in succession -- namely, if the cookie that partially controls whether to show the dialog isn't cleared between tests, the test fails This is true for our iPad tests, which run scenarios in succession, instead of in parallel.

### CODE REVIEW

**Comment by bencodeorg** [comment]
@islemaster Hamms is out, so could use another peek at this. 

New code is e635c10 through eb7a9f4 -- the rest are commits to try and get Drone to pass. The tests that were causing issues are passing when adding `test ios` and `test safari` -- however, we have a test that only succeeds when run in one browser at a time [per this PR](https://github.com/code-dot-org/code-dot-org/pull/21799), so it continued to fail until I got rid of all tags and ran only in Chrome.

This seems like a problem for anyone who wants to test multiple browsers in Drone, but apparently hasn't come up in the past 2.5 years, so maybe folks just don't use these tags.

File: apps/src/sites/studio/pages/layouts/_thank_donors_interstitial.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import firehoseClient from '@cdo/apps/lib/util/firehose';
 
 $(document).ready(function() {
   var already_shown = !!cookies.get('has_seen_thank_donors');
-  if (!already_shown && window.innerWidth > 720) {
+  if (!already_shown && window.innerWidth > 800 && window.innerHeight > 600) {
```
Comment (bencodeorg) [inline apps/src/sites/studio/pages/layouts/_thank_donors_interstitial.js:7]: The modal is ~700x500, so somewhat arbitrary decision to show on devices where there'd be a bit of padding. I want young kids to be able to dismiss it even if they can't read it, which will happen when they click/touch anywhere outside of the modal (or click/touch the button on the modal).

**Review by islemaster — COMMENTED** [review]
Will you please update the PR description with a short explanation of what motivated the revert in the first place, and what's changed this time around?  It will help whoever reads this code in the future.

Reading the diff, it looks like you've increased the minimum window size to show the modal and made some test changes.

**Comment by bencodeorg** [comment]
> Will you please update the PR description with a short explanation of what motivated the revert in the first place, and what's changed this time around? It will help whoever reads this code in the future.
> 
> Reading the diff, it looks like you've increased the minimum window size to show the modal and made some test changes.

Yes, apologies for not doing that the first time around -- done!

**Review by islemaster — APPROVED** [review]
Awesome, thank you.  Looks good!  I'm curious about that test you mentioned that only succeeds when running in one browser, since it looks like it's supposed to be annotated to avoid causing problems.  Maybe something that can be improved upon?  In any case, like you said it hasn't come up in years, so I don't think it's pressing to address.

**Comment by bencodeorg** [comment]
I _think_ the problem is here, that the test that doesn't succeed when test run in parallel still gets run on multiple browsers at once if `is_circle` (which I assume is `is_drone` now):

https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/test/ui/runner.rb#L654

**Pull request merged**